controller/DonationController.js

```diff
@@ -41,6 +41,42 @@ exports.makeDonation = catchAsyncError(async (req, res) => {
   }
 });
 
+/**
+ * @description Donate larger sums of money (NGN 500,001 and upwards)
+ * @route POST /v1/donation/bulk-sum
+ * @access PUBLIC
+ */
+exports.recordBulkDonation = catchAsyncError(async (req, res) => {
+  /**
+   * @todo Handle validation for req.body
+   */
+  const { name, email, phone_number, ...referralDetails } = req.body;
+  const newReferral = await Referral.createReferral({
+    ...referralDetails,
+    transactionCompleted: true
+  });
+
+  if (!newReferral) {
+    return res.status(400).json({
+      status: false,
+      message: "Could not record donation, possibly duplicate tx_ref"
```

Comment:
@Chiemezuo, do we want this whole flow to break if we cannot create an entry in the referral collection? I wouldn't think so. 

I think in the event of a failure, we want to fail silently, log a message and continue to create the actual donation.

Reply:
@edustipend-root The flow only breaks if it isn't a new referral. Referrals are only classified as "new" when their `tx_ref` (which is meant to be unique) has not already been saved. 
The intention is for the flow to break. To avoid duplicate transactions